### DISCUSSION TITLE
Problem pushing code even with ssh key

### DISCUSSION THREAD

**Discussion by alkaloge** [community/community#65441] [New to GitHub]
Hello!

I have been able to generate and upload the ssh key to settings/ssh key. However, when I do

>git push

and enter the ssh key (starting with SHA256:) I get this 
> remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.

Any ideas?

Thanks

**Comment by Fiesty-Cushion** [comment]
Ensure that your repository's remote URL is set to the SSH URL, which typically looks like `git@github.com:username/repository.git` You can check the remote URL using the following command:
`git remote -v`

If the URL is set to HTTPS URL, you can update it using:
`git remote set-url origin git@github.com:username/repository.git`

It should fix this issue but if it persists, i suggest you properly add the SSH key to github:
<https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account?platform=windows>
